test/real_content/test_swupd.bats

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env bats
+
+load "../functional/testlib"
+
+# Test swupd with real official content published on clearlinux.org
+# This test assumes all functional tests to be passing in order to work,
+# because we aren't going to test all swupd functionalities. The idea is
+# to test major swupd commands and standard use cases
+
+#TODO: Add test for crossing a format barrier
+
+# Optional parameters
+# $URL: Clearlinux content and version urls
+# $MAX_VERSIONS: The maximum number of versions we are going to be looking to test updates. Should be # at least 2 (current + update). Default is 15.
+
+local SWUPD_OPTS # Swupd options to use in tests
+#local FORMAT # Format of the last available version
+local VERSIONS # Array of last available versions. First item is the oldest
+local ROOT_DIR # Root directory of the installation
+
+# get_format: Echos the format of the version in $1
+# TODO: This is not the best way to handle that. we should have in swupd
+# codebase the information about the supported format of this swupd.
+get_format() {
+	v="$1"
+	curl -f ${URL}/${v}/format 2>/dev/null
+	return 0
+}
+
+# get_format_from_versions: Echos the first format available in a version from
+# a list of versions
+get_format_from_versions() {
+	for v in "$@"; do
+		format=$(get_format $v)
+		if [ -z "$format" ]; then # if version isn't published, discard
+			continue
+		else
+			echo $format
+			break
+		fi
+	done
+}
+
+test_setup() {
```

Comment:
I think some of the contents of test_setup only need to be run once and could be moved to the global_setup function.

Reply:
I agree, but I couldn't find an easy way to do that.
Is there a way to export an env variable from one test to the other?

Reply:
I see what you mean. The env variables don't persist between tests... Currently, I think the only ways to export variables between the tests are to call the set_env_variables function in setup() or to give them global scope. Maybe we could add a function to the testlib in the future to make this easier

Reply:
Now, I think this is fine as is. It would be useful to add a feature to testlib to solve this issue in the future.